addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3222,6 +3225,7 @@ var SnippetsMenu = Widget.extend({
                             self._disableUndroppableSnippets();
                             self.options.wysiwyg.odooEditor.unbreakableStepUnactive();
                             self.options.wysiwyg.odooEditor.historyStep();
+                            self.options.wysiwyg.options.onWysiwygChange && self.options.wysiwyg.options.onWysiwygChange();
```

Comment:
I'm very confused. The callback is called `onWysiwygChange` so in fact I would have expected it to be called by the wysiwyg when it changed, but it doesn't look to work like that at all so I'm not sure I understand what this callback even is anymore. It's very weird, in fact. We're passing a callback to wysiwyg but the wysiwyg itself don't even use it, it's just to make it callable by someone _else_ somewhere else down the line! How comes wysiwyg itself can't detect when a change is made and call the callback by itself ? If it's really not possible then I think the code here should be triggering something that the field might be listening to. That calls back to my previous comment. What about triggering `content_changed` here and making it so the field listens for `content_changed` to call `commitChanges` ? We must be careful about the other places where we might be triggering `content_changed`, of course.

Reply:
Maybe we need another event then ? Because right now it looks like we are passing a reference to a callback function from agent A to B except B doesn't really care about it, it's just so that we need it in C so we have to give it to B first so that it ultimately gives it to C.

Reply:
What do you mean by "we need another event" ? Also, what is your suggestion regarding who should define the callback function and how it is called?